pkg/operator/ceph/file/mds/mds.go

```diff
@@ -141,6 +141,20 @@ func (c *Cluster) Start() error {
 			return fmt.Errorf("failed to generate keyring for %q. %+v", resourceName, err)
 		}
 
+		// Check for existing deployment and set the daemon config flags
+		_, err = c.context.Clientset.AppsV1().Deployments(c.fs.Namespace).Get(mdsConfig.ResourceName, metav1.GetOptions{})
+		// We don't need to handle any error here
+		if err != nil {
+			// Set ceph mds config flag
```

Comment:
There could be other possible errors. I'd suggest checking also for `if errors.IsNotFound(err)`